src/app.rs

```diff
@@ -1166,6 +1526,36 @@ impl UiExtra for egui::Ui {
     }
 }
 
+fn get_entries_with_level<'a>(items: &Vec<&'a EntryID>, level: u64) -> Vec<Vec<&'a EntryID>> {
+    let mut split: Vec<Vec<&EntryID>> = vec![vec![]];
+    for entry in items {
+        let sublist = split.last_mut().unwrap();
+        match sublist.last_mut() {
+            Some(x) if entry.slot_index(level).unwrap() != x.slot_index(level).unwrap() => {
+                split.push(vec![entry]);
+            }
+            _ => sublist.push(entry),
+        }
+    }
+    split
+}
+
+fn get_filtered_entries<'a>(
```

Comment:
Similar to the above, trying to understand this code and not entirely sure what's being filtered for.